aws_lambda_powertools/utilities/parameters/base.py

```diff
@@ -145,7 +145,11 @@ def get_multiple(
 
         if transform is not None:
             for (key, value) in values.items():
-                values[key] = transform_value(value, transform, raise_on_transform_error)
+                _transform = get_transform_method(key, transform)
+                if _transform is None:
+                    continue
+
+                values[key] = transform_value(value, _transform, raise_on_transform_error)
```

Comment:
__Open question:__ Should we change the key name based on the transform?

E.g., let's say we add a yaml/json5/etc. transform in the future, what if someone wants to change a parameter from JSON to YAML but minimise code change?

Reply:
@nmoutschen  - in `auto` method, their key name is explicitly setting the transform method, then `.json` and `.json5` would have different extensions. And for new methods like `.xml`, `.properties` and  `.yaml` we would use different transform_methods.

Reply:
Sorry, I think I was not clear here so let me put an example.

Let's say I have a DynamoDB like this:

| id | sk | value |
|----|----|--------|
| my-param | param.json | {"message": "Hello"} |

I'm wondering, should the return value be A or B between:

```
a = {
    "param.json": {
        "message": "Hello"
    }
}
```

```
b = {
    "param": {
        "message": "Hello"
    }
}
```

Reply:
@nmoutschen  it should be the same as if all of them where `json` to be consistent with the `json` transform method.

Reply:
@nmoutschen i think it should be `a`, because if you turn off auto. The keys should be the same as before. 

But I see your point. If might be nicer for it to be `b` for the end user.

Reply:
@nmoutschen Actually I take that back it should be `a`. Because `b` might conflict when an existing value with the same name.

Reply:
@nmoutschen  as an example:

id | sk | value
-- | -- | --
my-param | param.json | {"message": "Hello"}
my-param | param| Plain text

Reply:
Good point! 👍 I agree with you on that. Since that's the current behaviour, no change needed here.